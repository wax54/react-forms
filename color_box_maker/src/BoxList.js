import React, {useState} from "react";
import { v4 as uuid } from 'uuid';

import Box from './Box';
import NewBoxForm from './NewBoxForm';

const BoxList = () => {
    const [ boxes, setBoxes ] = useState([{id: uuid(), width: 100, height:100, color: 'black'}]);
    const addNewBox = ({width, height, color}) => {
        setBoxes(boxes => [...boxes, { id: uuid(), width, height, color }])
    };
    const deleteBox = deleteId => {
        setBoxes( allBoxes => {
            return allBoxes.filter( ({id}) => id !== deleteId)
        })
    };

    return (
        <div className="BoxList">
            { 
            boxes.map(({id, width, height, color}) => 
                <Box key={id} id={id} width={width} height={height} color={color} deleteBox={deleteBox}/>)
            }
            <NewBoxForm onSubmit={addNewBox} /> 
        </div>
    )
};

export default BoxList;
