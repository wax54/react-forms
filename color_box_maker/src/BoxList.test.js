import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import BoxList from './BoxList';

it("Doesn't Blow up", () => {
    render(<BoxList />);
});

it('should match previous snapshot', function () {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});
