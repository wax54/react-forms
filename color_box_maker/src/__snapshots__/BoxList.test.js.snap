// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match previous snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form>
      <label
        for="height-input"
      >
        Height: 
      </label>
      <input
        id="height-input"
        min="0"
        name="height"
        required=""
        type="number"
        value=""
      />
      <label
        for="width-input"
      >
        Width: 
      </label>
      <input
        id="width-input"
        min="0"
        name="width"
        required=""
        type="number"
        value=""
      />
      <label
        for="color-input"
      >
        Color: 
      </label>
      <input
        id="color-input"
        name="color"
        required=""
        type="text"
        value=""
      />
      <button>
        Make me a Box!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
